### DISCUSSION TITLE
Behavior of declarative automation condition "since()"

### DISCUSSION THREAD

**Discussion by yamhirotoCS** [dagster-io/dagster#25179] [Q&A] [answered]
Hi, 
Can anyone help me understand the behaviour of AutomationCondition.since() in those examples below ?:

<img width="638" alt="Screenshot 2024-10-10 at 10 41 35" src="https://github.com/user-attachments/assets/07b97bb2-58a0-4f78-853f-109a62c146e5">

<img width="654" alt="Screenshot 2024-10-10 at 10 41 48" src="https://github.com/user-attachments/assets/11373caf-5fcf-400d-a669-fc6513dac762">

<img width="695" alt="Screenshot 2024-10-10 at 17 57 45" src="https://github.com/user-attachments/assets/5dbe40ab-27df-4c1d-ab17-8c65cf1fffa8">

Many thanks for your help !

**Comment by OwenKephart** [comment, marked as answer]
Hi @yamhirotoCS -- the `since()` condition is true if the top condition has become true more recently than the bottom condition.

So even if neither are true on a particular tick, the overall condition may still evaluate to true.

**Reply by yamhirotoCS** [reply]
hey @OwenKephart , thanks for coming back to me. But what about the two other cases?
and do you mean **top_condition.since(bottom_condition**)

**Reply by OwenKephart** [reply]
> do you mean top_condition.since(bottom_condition)

That's right (just using those terms as that's how they show up visually in the screenshots).

First image: If the conditions become true on the same tick, then the condition resolves to `False` (as they became true at the same time, the top condition did _not_ become true more recently than the bottom condition)

Second image: This indicates that on some tick, the top condition became true, and the bottom condition has not been true since that point in time, so the result is True.

Third image: On this tick, the top condition became true, and the bottom condition is still false, so the top condition became true more recently than the bottom condition, making the result True.

**Reply by yamhirotoCS** [reply]
Amazing, had to read your explanation a few times but it now makes more sense.
thanks again!

**Reply by OwenKephart** [reply]
No problem!
